templates/tfengine/components/cicd/envs/main.tf

```diff
@@ -12,64 +12,79 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-  run_on_push             = var.triggers.validate.run_on_push
   run_on_schedule         = var.triggers.validate.run_on_schedule
-  {{- if has . "cloud_source_repository"}}
   cloud_source_repository = var.cloud_source_repository
-  {{- end}}
-  {{- if has . "github"}}
   github                  = var.github
-  {{- end}}
   project_id              = var.project_id
   scheduler_region        = var.scheduler_region
   terraform_root          = var.terraform_root
   service_account_email   = var.service_account_email
+  
+  push = {
+    skip = var.triggers.validate.skip
+    disabled = !var.triggers.validate.run_on_push
+  }
+
+  scheduled = {
+    skip = var.triggers.validate.skip || var.triggers.validate.run_on_schedule == ""
+    disabled = true # Always disabled on push to branch.
+  }
 }
 
 module "plan_triggers" {
-  source                  = "./plan"
+  source                  = "./triggers"
 
-  env                    = var.env
+  command                 = "plan"
+  env                     = var.env
   branch_name             = var.branch_name
   managed_dirs            = var.managed_dirs
-  skip                    = var.triggers.plan.skip
-  run_on_push             = var.triggers.plan.run_on_push
   run_on_schedule         = var.triggers.plan.run_on_schedule
-  {{- if has . "cloud_source_repository"}}
   cloud_source_repository = var.cloud_source_repository
-  {{- end}}
-  {{- if has . "github"}}
   github                  = var.github
-  {{- end}}
   project_id              = var.project_id
   scheduler_region        = var.scheduler_region
   terraform_root          = var.terraform_root
   service_account_email   = var.service_account_email
+
+  push = {
+    skip = var.triggers.plan.skip
+    disabled = !var.triggers.plan.run_on_push
+  }
+
+  scheduled = {
+    skip = var.triggers.plan.skip || var.triggers.plan.run_on_schedule == ""
+    disabled = true # Always disabled on push to branch.
+  }
 }
 
 module "apply_triggers" {
-  source                  = "./apply"
+  source                  = "./triggers"
 
+  command                 = "apply"
   env                     = var.env
   branch_name             = var.branch_name
   managed_dirs            = var.managed_dirs
-  skip                    = var.triggers.apply.skip
-  run_on_push             = var.triggers.apply.run_on_push
   run_on_schedule         = var.triggers.apply.run_on_schedule
-  {{- if has . "cloud_source_repository"}}
   cloud_source_repository = var.cloud_source_repository
-  {{- end}}
-  {{- if has . "github"}}
   github                  = var.github
-  {{- end}}
   project_id              = var.project_id
   scheduler_region        = var.scheduler_region
   terraform_root          = var.terraform_root
   service_account_email   = var.service_account_email
+
+  push = {
+    skip = true # Always true since scheduled trigger may act as a push trigger too
+    disabled = true
+  }
+
+  scheduled = {
+    skip = var.apply.validate.skip
```

Comment:
Should this be var.triggers.apply?

Reply:
Thanks! That was a very obvious one and still didn't noticed